pkgs/development/libraries/g2o/default.nix

```diff
@@ -0,0 +1,22 @@
+{ lib, stdenv, fetchFromGitHub, cmake, eigen, suitesparse }:
+
+stdenv.mkDerivation {
+  pname = "g2o";
+  version = "2019-04-07";
```

Comment:
From the nixpkgs manual:

> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23". 

As you mentioned this software doesn't do releases so this might be debatable...